src/jobsautoreport/report.py

```diff
@@ -73,52 +120,49 @@ def __init__(self, querier: Querier):
     def _get_job_triggers_count(job_name: str, jobs: list[JobDetails]) -> int:
         return sum(1 for job in jobs if job.name == job_name)
 
-    def _get_job_states_count(
-        self, job_name: str, jobs: list[JobDetails]
-    ) -> JobStatesCount:
-        job_by_name = [job for job in jobs if job.name == job_name]
-        return self._get_states_count(job_by_name)
-
     @staticmethod
     def _get_number_of_jobs_by_state(
         jobs: list[JobDetails], job_state: JobState
     ) -> int:
         return len([job for job in jobs if job.state == job_state.value])
 
-    def _get_states_count(self, jobs: list[JobDetails]) -> JobStatesCount:
+    def _get_job_metrics(self, job_name: str, jobs: list[JobDetails]) -> JobMetrics:
+        job_by_name = [job for job in jobs if job.name == job_name]
+        return self._compute_job_metrics(job_by_name)
```

Comment:
```suggestion
    def _get_job_metrics(self, job_id: JobIdentifier, jobs: list[JobDetails]) -> IdentifiedJobMetrics:
        job_by_name = [job for job in jobs if job.name == job_id.name]
        return IdentifiedJobMetrics(job_id, self._compute_job_metrics(job_by_name))
```
Let's embrace the newly created types